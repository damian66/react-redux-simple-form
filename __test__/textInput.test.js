import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import ConnectedTextInput, { TextInput } from './../client/src/components/TextInput'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

import {createStore} from 'redux'

const fields = [{
    key: "email",
    name: "Email address",
    placeholder: "johndoe@gmail.com",
    type: "email",
    required: true,
    validate: () => "ok"
}];

// Snapshot
describe('>>>TextInput --- Snapshot',()=>{
    it('+++capturing Snapshot of TextInput', () => {                
        const renderedValue =  renderer.create(<TextInput id="email" type="text" fields={fields} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

/* ***** ***** ***** */

describe('>>>SnackBar --- Shallow Render REACT COMPONENTS',()=>{    
    const initialState = {
        fields: {
            list: fields
        }
    }
    const mockStore = configureStore()
    let store, wrapper

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = mount( <Provider store={store}><ConnectedTextInput id={initialState.fields.list[0].key} f /></Provider> )        
    })

    it('+++ render the component', () => {        
       expect(wrapper.length).toEqual(1);
    })
});
