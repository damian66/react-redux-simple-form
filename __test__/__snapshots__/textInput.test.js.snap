// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>TextInput --- Snapshot +++capturing Snapshot of TextInput 1`] = `
<div
  className="wrapper"
  tooltip={undefined}
>
  <input
    className="input"
    id="email"
    onBlur={[Function]}
    onChange={undefined}
    onFocus={undefined}
    placeholder={null}
    type="text"
    value={undefined}
  />
</div>
`;
